src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,145 @@
+        /// <param name="fieldRange">The field range goes hand in hand with the sensitivity of the magnetic sensor</param>
+        public void SetMode(Mode mode = Mode.CONTINUOUS,
+            Interrupt interrupt = Interrupt.DISABLE,
+            RollPointer rollPointer = RollPointer.DISABLE,
+            OutputRate outputRate = OutputRate.RATE_10HZ,
+            Oversampling oversampling = Oversampling.OS128,
+            FieldRange fieldRange = FieldRange.GAUSS_2)
+        {
+            Span<byte> interruptCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_2_ADDR,
+                (byte)rollPointer
+            };
+            Span<byte> resetCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_RESET_REG_ADDR,
+                0x01
+            };
+            byte config = (byte)((byte)mode | (byte)outputRate | (byte)fieldRange | (byte)oversampling);
+            Console.WriteLine(config);
+            Span<byte> setMainRegisteryCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_1_ADDR,
+                config
+            };
+            _i2cDevice.Write(interruptCommand);
+            _i2cDevice.Write(resetCommand);
+            _i2cDevice.Write(setMainRegisteryCommand);
+        }
+
+        /// <summary>
+        /// Reads status register to determine if data is avaliable.
+        /// </summary>
+        /// <returns>1 == read is ready. 0 == read is not ready</returns>
+        public bool IsReady()
+        {
+            _i2cDevice.WriteByte((byte)Registers.QMC_STATUS_REG_ADDR);
+            byte status = _i2cDevice.ReadByte();
+            Console.WriteLine(status);
+            return (status & 0x01) == 1;
+        }
+
+        /// <summary>
+        /// Read raw data from QMC5883L
+        /// </summary>
+        /// <returns>Raw Data</returns>
+        private Vector3 ReadDirectionVector()
+        {
+            Span<byte> xRead = stackalloc byte[2];
+            Span<byte> yRead = stackalloc byte[2];
+            Span<byte> zRead = stackalloc byte[2];
+
+            _i2cDevice.WriteByte((byte)Registers.QMC_X_LSB_REG_ADDR);
+            _i2cDevice.Read(xRead);
+            _i2cDevice.WriteByte((byte)Registers.QMC_Y_LSB_REG_ADDR);
+            _i2cDevice.Read(yRead);
+            _i2cDevice.WriteByte((byte)Registers.QMC_Z_LSB_REG_ADDR);
+            _i2cDevice.Read(zRead);
+
+            return new Vector3(BinaryPrimitives.ReadInt16BigEndian(xRead),
+                               BinaryPrimitives.ReadInt16BigEndian(yRead),
+                               BinaryPrimitives.ReadInt16BigEndian(zRead));
+        }
+
+        /// <summary>
+        /// Calculate heading
+        /// </summary>
+        /// <param name="vector">QMC5883L Direction Vector</param>
+        /// <returns>Heading (DEG)</returns>
+        private static double VectorToHeading(Vector3 vector)
```

Comment:
We're using UnitsNet for anything related to units. See https://github.com/dotnet/iot/tree/master/src/devices#using-units
You should use it for angles. So you the user then can decide what unit of angle they want.